python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -2151,7 +2149,7 @@ def check_changability(obj1, obj2):
     def _commit(self, objects=None):
         """ Helper method to unconditionally commit to the repository. The 'objects' list specifies objects
         to be commited (for example the subjobs). If objects are not specified then just the self is commited """
-
+        # TODO This function should force a flush. However it is currently used everywhere which causes too many flushes.
         self._getRoot()._setDirty()
```

Comment:
The purpose of `_commit` historically was to *force* a flush whenever something significant changed. This is why it use ubiquitously throughout `Job`. As you found and fixed, this was flushing far too often so the first thing we need to fix with `_commit` is reducing the places that it is called.